packages/grafana-ui/src/components/Sparkline/utils.ts

```diff
@@ -13,22 +13,24 @@ export function preparePlotFrame(sparkline: FieldSparkline, config?: FieldConfig
     ...config,
   };
 
-  return applyNullInsertThreshold({
-    frame: {
-      refId: 'sparkline',
-      fields: [
-        sparkline.x ?? {
-          name: '',
-          values: [...Array(length).keys()],
-          type: FieldType.number,
-          config: {},
-        },
-        {
-          ...sparkline.y,
-          config: yFieldConfig,
-        },
-      ],
-      length,
-    },
-  });
+  const frame = {
+    refId: 'sparkline',
+    fields: [
+      sparkline.x ?? {
+        name: '',
+        values: [...Array(length).keys()],
+        type: FieldType.number,
+        config: {},
+      },
+      {
+        ...sparkline.y,
+        config: yFieldConfig,
+      },
+    ],
+    length,
+  };
+
+  const sortedFrame = sortDataFrame(frame, 0);
```

Comment:
maybe do this only if we detect unsorted?

https://github.com/grafana/grafana/blob/b2fc285a5ddaa25265aea9a137fb37512ecdaf57/packages/grafana-data/src/transformations/transformers/joinDataFrames.ts#L354